rules/endpoints.go

```diff
@@ -308,17 +322,41 @@ func (r *DefaultRuleRenderer) endpointIptablesChain(
 		},
 	})
 
-	if dropEncap {
+	if dropEncap && r.Config.VXLANEnabled {
```

Comment:
I could possibly remove the `r.Config.VXLANEnabled ` if we just want behavior similar to before where we added the drop rules unconditionally leaving the tweaks to the actual the destination endpoint.

Reply:
Was wondering about this myself.  Is it possible to have a single cluster with mixed VXLAN/no VXLAN or IPIP/no IPIP usage?   If so then an individual node may have encapEnabled set to false but you still need to police the encap IPSet....

Feels like it might be safer.  In the normal case, if encapsulated is disabled then the IPSet will be empty anyways.

Reply:
Is there a reason we only want this with VXLAN enabled? I thought the rule tweaks were so that we could always have this rule in place and still allow the necessary traffic?

Reply:
I guess it's to make this work for IPv6 as well, where we don't have the IP set to match on?

Reply:
yeah, my vote would be to always program the rules, but just make them more specific so they only match on traffic from wep to a known Calico node.

For IPv4 vs IPv6, do we need separate rule sets actually? We don't support VXLAN or IPIP for v6 anyway.

Reply:
Yup, will update to always program the rule.

Reply:
I was looking into why the FVs started to fail after removing the check to only program the rules if the VXLAN config is enabled and it turns out that Felix only creates and maintains the corresponding host ipset if IPIP or VXLAN is enabled. 

I think a few options are:

Option 1: 
Keep the logic for only enabling the encap spoofing deny rules, which will prevent us from dropping legitimate encap traffic on hosts where encap is disabled. If encap is enabled and traffic is ipv4 then the ipset based rule will refine the traffic we drop based on destination IP check.

Option 2:
Create an empty ipset even if the encap method is disabled. While this will fix the FV's, this may not prevent the bug of incorrectly blocking legitimate encap traffic. For the fix to actually work, we may have to start maintaining a list of calico hosts and add this to the rule instead? The current fix uses specific encap related ipsets, and these may not be the same as a Calico host?